core-bundle/contao/library/Contao/Image.php

```diff
@@ -105,9 +104,18 @@ public static function getPath($src)
 			trigger_deprecation('contao/core-bundle', '5.2', 'Using the "%s" icon has been deprecated and will no longer work in Contao 6. Use the "%s--disabled" icon instead.', $filename, substr($filename, 0, -1));
 		}
 
-		$projectDir = System::getContainer()->getParameter('kernel.project_dir');
+		// Use path from icon manifest
+		$icons = System::getContainer()->getParameter('contao.backend.icons');
+
+		if (null !== ($icon = ($icons["$filename.svg"] ?? null)))
+		{
+			return ltrim($icon['path'], '/');
+		}
```

Comment:
Isn't this less complex?

```suggestion
		if (isset($icons["$filename.svg"]))
		{
			return ltrim($icons["$filename.svg"]['path'], '/');
		}
```

Reply:
Both fine for me, I think that is mostly CS. :slightly_smiling_face: